packages/financial-templates-lib/src/clients/OptimisticOracleClient.ts

```diff
@@ -0,0 +1,249 @@
+  private unproposedPriceRequests: RequestPrice[] = [];
+  private undisputedProposals: ProposePrice[] = [];
+  private expiredProposals: ProposePrice[] = [];
+  private settleableDisputes: DisputePrice[] = [];
+
+  /**
+   * @notice Constructs new OptimisticOracleClient.
+   * @param {Object} logger Winston module used to send logs.
+   * @param {Object} oracleAbi OptimisticOracle truffle ABI object.
+   * @param {Object} votingAbi Voting truffle ABI object.
+   * @param {Object} web3 Provider from Truffle instance to connect to Ethereum network.
+   * @param {String} oracleAddress Ethereum address of the OptimisticOracle contract deployed on the current network.
+   * @param {String} votingAddress Ethereum address of the Voting contract deployed on the current network.
+   * @param {Number} lookback Any requests, proposals, or disputes that occurred prior to this timestamp will be ignored.
+   * Used to limit the web3 requests made by this client.
+   * @return None or throws an Error.
+   */
+  constructor(
+    private readonly logger: Logger,
+    oracleAbi: Abi,
+    votingAbi: Abi,
+    public readonly web3: Web3,
+    oracleAddress: string,
+    votingAddress: string,
+    public readonly lookback = 604800 // 1 Week
+  ) {
+    // Optimistic Oracle contract:
+    this.oracle = (new web3.eth.Contract(oracleAbi, oracleAddress) as unknown) as OptimisticOracleWeb3;
+
+    // Voting contract we'll use to determine whether OO disputes can be settled:
+    this.voting = (new web3.eth.Contract(votingAbi, votingAddress) as unknown) as VotingAncillaryInterfaceTestingWeb3;
+
+    // Oracle Data structures & values to enable synchronous returns of the state seen by the client.
+    this.unproposedPriceRequests = [];
+    this.undisputedProposals = [];
+    this.expiredProposals = [];
+    this.settleableDisputes = [];
+  }
+
+  // Returns an array of Price Requests that have no proposals yet.
+  public getUnproposedPriceRequests(): RequestPrice[] {
+    return this.unproposedPriceRequests;
+  }
+
+  // Returns an array of Price Proposals that have not been disputed.
+  public getUndisputedProposals(): ProposePrice[] {
+    return this.undisputedProposals;
+  }
+
+  // Returns an array of expired Price Proposals that can be settled and that involved
+  // the caller as the proposer
+  public getSettleableProposals(caller: string): ProposePrice[] {
+    return this.expiredProposals.filter((event) => {
+      return event.proposer === caller;
+    });
+  }
+
+  // Returns disputes that can be settled and that involved the caller as the disputer
+  public getSettleableDisputes(caller: string): DisputePrice[] {
+    return this.settleableDisputes.filter((event) => {
+      return event.disputer === caller;
+    });
+  }
+
+  // Returns the last update timestamp.
+  public getLastUpdateTime(): number {
+    return this.lastUpdateTimestamp;
+  }
+
+  private _getPriceRequestKey(
```

Comment:
this took an already messy method and made it look even scarier with the type included. not sure if there is a better way to lint this. I guess if the `OptimisticOracleWeb3Events` variable name was shorter this might be able to fit on one line?

Reply:
I made an aliased type. Let me know if you like this better.

Reply:
looks much better now, thanks!